frontend/src/metabase/common/components/Table/Table.tsx

```diff
@@ -1,69 +1,64 @@
+import type { PropsWithChildren, ReactNode } from "react";
 import React from "react";
 
-import { color } from "metabase/lib/colors";
 import { Box, Flex, Icon } from "metabase/ui";
 
-type BaseRow = Record<string, unknown> & { id: number | string };
-
-type ColumnItem = {
-  name: string;
-  key: string;
-};
+import type { BaseRow, ColumnItem } from "./types";
+import { useTableSorting } from "./useTableSorting";
 
 export type TableProps<Row extends BaseRow> = {
   columns: ColumnItem[];
   rows: Row[];
-  rowRenderer: (row: Row) => React.ReactNode;
-  tableProps?: React.HTMLAttributes<HTMLTableElement>;
-};
+  rowRenderer: (row: Row) => ReactNode;
+  formatValueForSorting?: (row: Row, columnName: string) => any;
+  defaultSortColumn?: string;
+  defaultSortDirection?: "asc" | "desc";
+  appendToBody?: ReactNode;
```

Comment:
This seems an odd parameter to me. Why would we need to append elements to the table body?

Reply:
Can we add a something to the row renderer so that it knows to render an empty row or something? Otherwise, the old `isEmpty` prop you had before probably makes more sense